### PR TITLE
fix(helm): add nodeselector and tolerations to init job

### PR DESCRIPTION
**Pull request by jakedsouza**
### SUMMARY

- Fixes issue #18004 adding a node selector and toleration to the init job configurable in the values file. 

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
N/A

### TESTING INSTRUCTIONS
- Set the nodeselector, toleration and affinity to the `init` block in the `values.yaml` file. 
```
  ## Node selector to be added to init job pods
  nodeSelector:
    role: test-nodeselector
  ## Tolerations to be added to init job pods
  tolerations:
      - key: role
        operator: Equal
        value: test-toleration
  ## Affinity to be added to init job pods
  affinity: {}
```

### ADDITIONAL INFORMATION

- [x] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Comment by jakedsouza** [comment]
@nytai @zhaoyongjie @craig-rueda the linter is failing due to a chart version mismatch, should I bump the version to `0.5.6`

**Review by craig-rueda — APPROVED** [review]
LGTM - pls bump chart version

**Review by zhaoyongjie — APPROVED** [review]

File: helm/superset/values.yaml [hunk 1]
```diff
@@ -347,6 +347,12 @@ init:
     fi
   ## Annotations to be added to init job pods
   podAnnotations: {}
+  ## Node selector to be added to init job pods
+  nodeSelector: {}
+  ## Tolerations to be added to init job pods
+  tolerations: []  
+  ## Affinity to be added to init job pods
+  affinity: {}
```
Comment (ad-m) [inline helm/superset/values.yaml:355]: Could you rebase and update `values.schema.json` too?

https://github.com/apache/superset/blob/4e2bdd49c4f23ace04ccde652fc6dfc8a4a24b0d/helm/superset/values.schema.json#L312-L378

**Review by ad-m — CHANGES_REQUESTED** [review]

**Comment by olahouze** [comment]
Hello

I have exactly the same problem

Can you apply this modification in the next version of the chart?

**Comment by NephilimJaeger** [comment]
Im with the same problem :(

**Review by craig-rueda — APPROVED** [review]
Lgtm

**Comment by eschutho** [comment]
It looks like this PR still needs a version bump.

**Comment by EugeneTorap** [comment]
@zhaoyongjie Can we merge it?

**Comment by losblancoo** [comment]
@jakedsouza why was the PR closed?

**Comment by asayeb** [comment]
I have applied the toleration and nodeSelector to the pods. all pods have the desired toleration and nodeSelector except for Redis pod. I haven't set up any caching config. it is deployed as it is in the 'values.yaml' file. What needs to be done to apply these setting to Redis pod?

**Comment by maximdogonov** [comment]
Hi, 

I setup: 
nodeSelector:
      nodetype: core-ccx23

but it applies only for superset and worker. 

Redis and Postgresql with out node-selector. 

How to fix that?

**Pull request closed without merging**